src/main/java/com/kodilla/ecommercee/mapper/PaymentMapper.java

```diff
@@ -1,21 +1,39 @@
 package com.kodilla.ecommercee.mapper;
 
 import com.kodilla.ecommercee.domain.Payment;
+import com.kodilla.ecommercee.domain.UserOrder;
 import com.kodilla.ecommercee.dto.PaymentDto;
+import com.kodilla.ecommercee.exception.UserOrderNotFoundException;
+import com.kodilla.ecommercee.repository.UserOrderRepository;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
+import javax.jws.soap.SOAPBinding;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
 @Component
 public class PaymentMapper {
+    @Autowired
+    private UserOrderRepository userOrderRepository;
 
-    public Payment mapToPayment(final PaymentDto paymentDto) {
-        return new Payment(
+    public Payment toPayment(final PaymentDto paymentDto) throws UserOrderNotFoundException {
+        Payment payment = new Payment(
                 paymentDto.getId(),
                 paymentDto.getPrice(),
                 paymentDto.isPaymentStatus()
         );
+        try {
+            payment.setUserOrder(userOrderRepository.findById(paymentDto.getUserOrderId()).get());
```

Comment:
Taka obsługa Optionala nie jest dobrym pomysłem. Zamiast czekać aż rzuci wyjątek, lapać go i rzucać kolejny, to powinniśmy wywołać metodę isPresent żeby sprawdzić czy znajduje się tam wartość i jeśli nie ma to rzucić wyjątek.
Inna opcja to zamiast metody get() użyć orElseThrow i tam podać wyjątek jako parametr

Reply:
Poprawiłam to + przetestowałam w Postmanie wszystkie metody i prawidłowość rzucania wyjątków.